gateway/api/services/result_storage.py

```diff
@@ -0,0 +1,78 @@
+"""
+This module handle the access to the result store
+"""
+import os
+import logging
+import mimetypes
+from typing import Optional, Tuple
+from wsgiref.util import FileWrapper
+from django.conf import settings
+
+logger = logging.getLogger("gateway")
+
+
+class ResultStorage:
+    """Handles the storage and retrieval of user job results."""
+
+    RESULT_FILE_EXTENSION = ".json"
+    ENCODING = "utf-8"
+
+    def __init__(self, username: str):
+        """Initialize the storage path for a given user."""
+        self.user_results_directory = os.path.join(
+            settings.MEDIA_ROOT, username, "results"
+        )
+        os.makedirs(self.user_results_directory, exist_ok=True)
+
+    def __build_result_path(self, job_id: str) -> str:
```

Comment:
@paaragon to maintain the same semantic as we had in `FileStorage` let's use `__get_result_path` instead of build, WDYT?